app/controllers/account/admin/billing_contracts_controller.rb

```diff
@@ -3,14 +3,15 @@ class Account::Admin::BillingContractsController < Account::Admin::AdminControll
   before_action :set_company
   before_action :set_billing_contract, only: %i[show edit update destroy]
 
-  def index
+  def index # rubocop:disable Metrics/AbcSize
     authorize :billing_contract
     @billing_contracts = @company.billing_contracts.page params[:page]
     respond_to do |format|
       format.xlsx do
+        @billing_contracts = @company.billing_contracts.all
         response.headers['Content-Disposition'] = "attachment; filename=billing-contracts.xlsx"
       end
-      format.csv { send_data @billing_contracts.to_csv }
+      format.csv { send_data @company.billing_contracts.all.to_csv }
```

Comment:
how about applying page pagination only in format.html?
 and have @billing_contracts ready for xlsx, csv, html the same

Reply:
it's a really great idea!